src/main/java/blitz/rijks/RijksSearchFrame.java

```diff
@@ -0,0 +1,147 @@
+package blitz.rijks;
+
+import com.andrewoid.ApiKey;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import javax.imageio.*;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.io.IOException;
+import java.net.URL;
+
+public class RijksSearchFrame extends JFrame {
+
+    public JButton prevPageButton = new JButton("Previous Page");
+    public JButton nextPageButton = new JButton("Next Page");
+    public JTextField search = new JTextField(80);
+    public JPanel resultPanel = new JPanel();
+    public int currentPage = 1;
+    private final RijksService service;
+
+
+    public RijksSearchFrame() {
+        setTitle("Museum Frame");
+        setSize(800, 600);
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        JPanel topPanel = new JPanel();
+        topPanel.setLayout(new BorderLayout());
+
+        JPanel searchPanel = new JPanel();
+        searchPanel.setLayout(new FlowLayout(FlowLayout.CENTER, 10, 10));
+        searchPanel.add(prevPageButton);
+        searchPanel.add(search);
+        searchPanel.add(nextPageButton);
+
+        JScrollPane resultScrollPane = new JScrollPane(resultPanel);
+        topPanel.add(searchPanel, BorderLayout.CENTER);
+
+        add(topPanel, BorderLayout.NORTH);
+        add(resultScrollPane, BorderLayout.CENTER);
+
+        prevPageButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (currentPage > 1) {
+                    currentPage--;
+                    searchArtworks();
+                }
+            }
+        });
+
+        nextPageButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                currentPage++;
+                searchArtworks();
+            }
+        });
+
+        search.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                currentPage = 1;
+                searchArtworks();
+            }
+        });
+
+        resultPanel.setLayout(new GridLayout(0, 4, 10, 10));
+        service = new RijksServiceFactory().getService();
+        searchArtworks();
+    }
+
+    private void searchArtworks() {
+        ApiKey apiKey = new ApiKey();
+        String query = search.getText().trim();
+        if (query.isEmpty()) {
+            service.getPage(apiKey.get(), currentPage)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.from(SwingUtilities::invokeLater))
```

Comment:
This isn't how I said to do it.
Where did you learn it this way?
(I might like it better)

Reply:
I honestly don't remember. I think I just assumed that's how we were supposed to do it. Should I change it to your way?